### DISCUSSION TITLE
User onboarding: Wagtail tour guides

### DISCUSSION THREAD

**Discussion by thibaudcolas** [wagtail/wagtail#9474] [Ideas]
Guided tours are a proposed new feature in Wagtail. Discussing this with the core team about a month ago, the overall feedback was that although this could be useful it likely wouldn’t be a good fit for Wagtail core – so we moved on with a new Help menu (#9351).

I thought I’d still share our ideas in this space here, for future reference / implementation as an external package.

## Features

- Remember completion state so we don’t offer the same guide twice (covered by work for #9351)
- Support multiple guides
- Define guides with code
- Define guides within CMS UI
- Multilingual tour guides
- User progression tracking / reporting
- Guide tour selection UI for users
- Core feature or third-party package

### Shepherd

<https://shepherdjs.dev/>

-   MIT license
-   Used by Drupal in core
-   Used by LB for <https://dev.to/lb/how-to-build-an-interactive-guide-for-users-in-the-wagtail-cms-admin-2dcp> 

### Intro.js

<https://introjs.com/>

-   AGPL
-   Drupal & WordPress integrations

### Rejected

-   <http://heelhook.github.io/chardin.js/> (old)
-   <http://tracelytics.github.io/pageguide/> (old)
-   <https://github.com/LinkedInAttic/hopscotch> (old)
-   <https://github.com/gilbarbara/react-joyride> (too much react)
-   <https://ejulianova.github.io/guides/> (too basic)
-   <https://github.com/LikaloLLC/tourguide.js/> (too basic)
-   <https://github.com/socialtables/react-user-tour> (too basic)
-   <https://github.com/elrumordelaluz/reactour> (problematic keyboard support) (used by Kalob)

### GuideChimp

<https://github.com/Labs64/GuideChimp>

-   Browser extension (prototype) (<https://github.com/Labs64/GuideChimp-tours>)
-   GPL-style license

### Tutorial from LB

<https://dev.to/lb/how-to-build-an-interactive-guide-for-users-in-the-wagtail-cms-admin-2dcp> 

### wagtail-onboarding Wagtail 2.12 package

https://pypi.org/project/wagtail-onboarding/

**Comment by lb-** [comment]
I think this is a really great idea but also agree that this is better served as an external package. 

If this gets built, it would be great to leverage the existing Wagtail Guide content that gets built on that site.

See https://github.com/wagtail/guide/issues/160 for some ideas on how that guide could provide it's content in a versioned/fixed format way.

This may be a stretch but even the guide project itself, which is a website, could also become a package to be included within Wagtail and parts of it be automatically shown as an onboarding guide.

This may not really be practical but was kind of the original inspiration for the issue that became the guide in the first place. 

https://github.com/wagtail/wagtail/discussions/7824
